src/shogun/classifier/mkl/MKL.cpp

```diff
@@ -433,12 +433,6 @@ bool CMKL::train_machine(CFeatures* data)
 
 	if (interleaved_optimization)
 	{
-		if (svm->get_classifier_type() != CT_LIGHT && svm->get_classifier_type() != CT_SVRLIGHT)
-		{
-			SG_ERROR("Interleaved MKL optimization is currently "
-					"only supported with SVMlight\n");
-		}
-
```

Comment:
Multikernel classes are the only classes that make use of the machine type. They use it for error checking.

Reply:
So what is the replacement here?

Reply:
Dynamic casting would solve the problem, but I think it might not be the best solution since it breaks abstraction. Adding specific enum for SVMs would work, but it might defeat the point of removing machine type.

Reply:
Add a virtual method `bool CSVM::supports_mkl()` that returns `false` by default, and then `CSVMLight` overrides this returning true. Then you can all that method

Reply:
The problem here is whether MKL can make interleaving optimization using a certain svm class. I'm not sure, but this seems like information concerning the mkl class, not the svm itself, judging from the error message.

Reply:
No matter how the direction is, we will need to add something to the API/parameters of those classes to communicate. See also my other comment

Reply:
Sorry, bear with me. I still think that something like `bool CSVM::supports_mkl()` implemented in derived classes is not the best way. Maybe we could add a property like `SVMType` and let the MKL class has a list of types that he supports this optimization on.
So for example, if MKL is a separate plugin, it can be updated to support more types without bothering the other classes.
I don't have a clear picture about this plugin interface, so if what I say doesn't make sense, say so :"D